drivers/aks/aks_driver.go

```diff
@@ -92,219 +120,395 @@ func (d *Driver) GetDriverCreateOptions(ctx context.Context) (*types.DriverFlags
+	state.NetworkPodCIDR = options.GetValueFromDriverOptions(driverOptions, types.StringType, "pod-cidr", "podCidr").(string)
+	state.NetworkServiceCIDR = options.GetValueFromDriverOptions(driverOptions, types.StringType, "service-cidr", "serviceCidr").(string)
+
+	state.Location = options.GetValueFromDriverOptions(driverOptions, types.StringType, "location").(string)
+	state.DNSPrefix = options.GetValueFromDriverOptions(driverOptions, types.StringType, "master-dns-prefix", "masterDnsPrefix").(string)
+	state.KubernetesVersion = options.GetValueFromDriverOptions(driverOptions, types.StringType, "kubernetes-version", "kubernetesVersion").(string)
+	state.Tags = make(map[string]string)
+	tagValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "tags").(*types.StringSlice)
 	for _, part := range tagValues.Value {
 		kv := strings.Split(part, "=")
 		if len(kv) == 2 {
-			state.Tag[kv[0]] = kv[1]
+			state.Tags[kv[0]] = kv[1]
 		}
 	}
-	return state, state.validate()
+
+	state.BaseURL = options.GetValueFromDriverOptions(driverOptions, types.StringType, "base-url").(string)
+	state.ClientID = options.GetValueFromDriverOptions(driverOptions, types.StringType, "client-id", "clientId").(string)
+	state.ClientSecret = options.GetValueFromDriverOptions(driverOptions, types.StringType, "client-secret", "clientSecret").(string)
+	state.TenantID = options.GetValueFromDriverOptions(driverOptions, types.StringType, "tenant-id", "tenantId").(string)
+	state.AuthBaseURL = options.GetValueFromDriverOptions(driverOptions, types.StringType, "auth-base-url", "authBaseUrl").(string)
+
+	state.DisplayName = options.GetValueFromDriverOptions(driverOptions, types.StringType, "display-name", "displayName").(string)
+
+	return &state, state.validate(isCreating)
 }
 
-func (state *state) validate() error {
-	if state.Name == "" {
-		return fmt.Errorf("cluster name is required")
+func (state *state) validate(isCreating bool) error {
+	if state.SubscriptionID == "" {
+		return fmt.Errorf(`"subscription id" is required`)
 	}
 
 	if state.ResourceGroup == "" {
-		return fmt.Errorf("resource group is required")
-	}
-
-	if state.SSHPublicKeyPath == "" && state.SSHPublicKeyContents == "" {
-		return fmt.Errorf("path to ssh public key or public key contents is required")
+		return fmt.Errorf(`"resource group" is required`)
 	}
 
 	if state.ClientID == "" {
-		return fmt.Errorf("client id is required")
+		return fmt.Errorf(`"client id" is required`)
 	}
 
 	if state.ClientSecret == "" {
-		return fmt.Errorf("client secret is required")
+		return fmt.Errorf(`"client secret" is required`)
 	}
 
-	if state.SubscriptionID == "" {
-		return fmt.Errorf("subscription id is required")
+	if isCreating {
+		if state.LinuxSSHPublicKeyContents == "" && state.LinuxSSHPublicKeyPath == "" {
+			return fmt.Errorf(`"ssh public key contents or path" is required`)
+		}
+	} else {
+		if state.Location == "" {
```

Comment:
is `location` required for both creating and updating?

Reply:
`location` filed is a requirement for the request and the following code has injected it:
https://github.com/rancher/kontainer-engine/blob/f5495908e7f8225d301316fd78a8fd3dfec803d2/drivers/aks/aks_driver.go#L601-L603